dpypelines/s3_tar_received.py

```diff
@@ -67,12 +67,23 @@ def start(s3_object_name: str):
     # get_dataset_id() currently empty - returns "not-specified"
     # To be updated once we know where the dataset_id can be extracted from (not necessarily s3_object_name as suggested by argument name)
     try:
-        dataset_id = get_dataset_id(s3_object_name)
+        manifest_dict = local_store.get_lone_matching_json_as_dict("manifest.json")
         logger.info(
-            "Successfully retrieved dataset_id", data={"dataset_id": dataset_id}
+            "Successfully retrieved dataset_id", data={"files_found": local_store.get_file_names(),
+             "pattern_looked_for": "manifest.json"}
         )
     except Exception as err:
-        logger.error("Failed to retrieve dataset_id", err)
+        logger.error("Failed to to retrieve file: manifest.json", err)
+        de_notifier.failure()
+        raise err
+    
+    try:
+        dataset_id = get_dataset_id(manifest_dict)
```

Comment:
Change `dataset_id` to `source_id` (and in the logger statements on L94 and L99)